### DISCUSSION TITLE
Icon as axis value

### DISCUSSION THREAD

**Discussion by nicolasthemmer-aaa** [patrykandpatrick/vico#667] [Questions] [answered]
I need to create an icon for values on the x-axis for grouped columns. The valueFormatter only accepts CartesianValueFormatter. Is there a workaround for that issue that anybody can think of?

Any help would be appreciated 🙏 

<img width="76" alt="image" src="https://github.com/patrykandpatrick/vico/assets/13311585/96e53b07-ded1-44c7-b4f4-3fab26cb114c">

**Comment by patrickmichalik** [comment, marked as answer]
Hello! Since the return type of `CartesianValueFormatter#format` is `CharSequence`, you can implement this by using `SpannableString` and `ImageSpan` or `DynamicDrawableSpan`.

**Reply by nicolasthemmer-aaa** [reply]
Thank you very much, I did not think of that 🙏

**Reply by patrickmichalik** [reply]
You’re welcome!
